scripts/build_desktop_app_with_firebase.py

```diff
@@ -0,0 +1,281 @@
+  """
+  triplet_name = [arch]
+  if platform.system() == 'Windows':
+    triplet_name.append('windows')
+    triplet_name.append('static')
+    if msvc_runtime_library == 'dynamic':
+      triplet_name.append('md')
+  elif platform.system() == 'Darwin':
+    triplet_name.append('osx')
+  elif platform.system() == 'Linux':
+    triplet_name.append('linux')
+  else:
+    raise ValueError("Unsupported platform. "
+                     "This function works only on Windows, Linux or Mac platforms.")
+
+  triplet_name = '-'.join(triplet_name)
+  print("Using vcpkg triplet: {0}".format(triplet_name))
+  return triplet_name
+
+def build_source_vcpkg_dependencies(sdk_source_dir, arch, msvc_runtime_library):
+  """Build C++ dependencies for firebase source SDK using vcpkg.
+
+  Args:
+   sdk_source_dir (str): Path to Firebase C++ source directory.
+   arch (str): Platform Architecture (example: 'x64').
+   msvc_runtime_library (str): Runtime library for MSVC (eg: 'static', 'dynamic').
+  """
+  # TODO: Remove this once dev branch of firebase-cpp-sdk repo has been merged
+  # onto main branch. This is required because vcpkg lives only in dev branch currently.
+  subprocess.run(['git', 'checkout', 'dev'],
+                 cwd=sdk_source_dir, check=True)
+  subprocess.run(['git', 'pull'], cwd=sdk_source_dir, check=True)
+
+  python_exe = 'python3' if distutils.spawn.find_executable('python3') else 'python'
+  subprocess.run([python_exe, "scripts/gha/install_prereqs_desktop.py"],
+                 cwd=sdk_source_dir, check=True)
+  subprocess.run([python_exe, "scripts/gha/build_desktop.py", "--arch", arch,
+                              "--msvc_runtime_library", msvc_runtime_library,
+                              "--vcpkg_step_only"], cwd=sdk_source_dir, check=True)
+
+def build_app_with_source(app_dir, sdk_source_dir, build_dir, arch,
+                          msvc_runtime_library='static', config=None,
+                          target_format=None):
+  """Build desktop app directly against the firebase C++ SDK source.
+
+  Since this invovles a cmake configure, it is advised to run this on a clean
+  build directory.
+
+  Args:
+   app_dir (str): Path to directory containing application's CMakeLists.txt.
+   sdk_source_dir (str): Path to firebase C++ SDK source directory (root of github repo).
+   build_dir (str): Output build directory.
+   arch (str): Platform Architecture (example: 'x64').
+   msvc_runtime_library (str): Runtime library for MSVC (eg: 'static', 'dynamic').
+   config (str): Release/Debug config.
+          If its not specified, cmake's default is used (most likely Debug).
+   target_format (str): If specified, build for this targetformat ('frameworks' or 'libraries').
+  """
+  # Cmake configure
+  cmd = ['cmake', '-S', '.', '-B', build_dir]
+  cmd.append('-DFIREBASE_CPP_SDK_DIR={0}'.format(sdk_source_dir))
+
+  # If generator is not specifed, default for platform is used by cmake, else
+  # use the specified value
```

Comment:
Nit:  [The Google style guide ](https://google.github.io/styleguide/pyguide.html#386-punctuation-spelling-and-grammar)says that all comments should include punctuation.

Reply:
Good to know. Thanks for pointing this out. Done.